openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/storages/AveragedMultiTreeSPEntry.java

```diff
@@ -0,0 +1,72 @@
+/*  This file is part of Openrouteservice.
+ *
+ *  Openrouteservice is free software; you can redistribute it and/or modify it under the terms of the
+ *  GNU Lesser General Public License as published by the Free Software Foundation; either version 2.1
+ *  of the License, or (at your option) any later version.
+
+ *  This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Lesser General Public License for more details.
+
+ *  You should have received a copy of the GNU Lesser General Public License along with this library;
+ *  if not, see <https://www.gnu.org/licenses/>.
+ */
+package org.heigit.ors.routing.graphhopper.extensions.storages;
+
+import java.util.Arrays;
+
+/**
+ * Shortest Path Tree with multiple fields for containing multiple source information.
+ * The weight is the average of all non-infinite entries for order in priority queue.
+ * Average weight can result in better performance than minimum weight.
+ * @author Hendrik Leuschner
+ *
+ */
+public class AveragedMultiTreeSPEntry extends MultiTreeSPEntry {
+	private double weight = Double.POSITIVE_INFINITY;
+
+	public AveragedMultiTreeSPEntry(int adjNode, int edgeId, double edgeWeight, boolean updated, AveragedMultiTreeSPEntry parent, int numTrees) {
+		super(adjNode,edgeId, edgeWeight, updated, parent, numTrees);
+		updateWeights();
+	}
+
+	public void setSubItemOriginalEdgeIds(int newOriginalEdgeId) {
+		for (MultiTreeSPEntryItem item : items) {
+			item.setOriginalEdge(newOriginalEdgeId);
+		}
+	}
+
+
+	@Override
+	public void updateWeights() {
+		double averageWeight = 0;
+		int numNonInfiniteWeights = 0;
+		for (int i = 0; i < items.length; i++) {
+			MultiTreeSPEntryItem item = items[i];
```

Comment:
Calling `item.getWeight()` twice could be avoided by replacing the `MultiTreeSPEntryItem item` by `double weight = items[i].getWeight();`.

Reply:
This will not have any optimizing effect (its only a call to a field, and will be optimized by the Compiler anyway). Do you think it's better for readability? I don't think so, but if you like, I can change it.

Reply:
A agree that in this particular case performance-wise it probably makes no difference at all. However, I think it's in general better to avoid subsequent calls of the same method if the result does not change and can be stored between the calls.

Reply:
done